src/index.ts

```diff
@@ -677,7 +683,7 @@ export class Pool extends (EventEmitter as new () => PoolEmitter) {
       this.connections.splice(connectionIndex, 1);
     }
 
-    client.end().catch((ex) => {
+    await client.end().catch((ex) => {
```

Comment:
Can you please change this to a try/catch block?
```ts
try {
  await client.end();
} catch (ex) {
  const { message } = ex as Error;
  if (!/This socket has been ended by the other party/giu.test(message)) {
    this.emit('error', ex);
  }
}
```

Reply:
Will do.